src/api/pool.go

```diff
@@ -0,0 +1,109 @@
+package api
+
+import (
+	"context"
+	"github.com/go-resty/resty/v2"
+	"golang.org/x/sync/errgroup"
+	"runtime"
+	"sync"
+)
+
+type Pool struct {
+	client          *Client         // http client
+	ctx             context.Context // context of the parallel work
+	workers         *errgroup.Group // error group -> if one worker fails, all will be stopped
```

Comment:
`context` a `errgroup` su synchronizacne nastroje medzi `goroutines` (vlaknami).

Funguje to tak, ze v `errgroup` su nejake vlakna ... a ak v jednom nastane chyba, tak sa ukoncia vsetky.